src/main/java/org/opensrp/web/utils/MaskingUtils.java

```diff
@@ -0,0 +1,167 @@
+package org.opensrp.web.utils;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+import org.joda.time.DateTime;
+import org.joda.time.LocalDate;
+import org.smartregister.domain.Address;
+import org.smartregister.domain.Client;
+
+public class MaskingUtils {
+	
+	/**
+	 * Expects date in the ISO 8601 format e.g. 1980-11-02
+	 * 
+	 * @param date String
+	 * @return Masked date e.g. 1980-01-01
+	 */
+	public String maskDate(String date) {
+		
+		if (StringUtils.isNotBlank(date) && StringUtils.contains(date, '-')) {
+			
+			return new StringBuilder(date.substring(0, date.indexOf('-'))).append("-01-01").toString();
+			
+		}
+		
+		return "";
+		
+	}
+	
+	/**
+	 * @param date Date object
+	 * @return Masked date e.g. 1980-01-01
+	 */
+	public Date maskDate(Date date) {
+		
+		return date != null ? maskDateCore(date) : null;
+		
+	}
+	
+	/**
+	 * @param date joda LocalDate object
+	 * @return Masked LocalDate e.g. 1980-01-01
+	 */
+	public LocalDate maskDate(LocalDate date) {
+		
+		return date != null ? new LocalDate(maskDateCore(date.toDate())) : null;
+		
+	}
+	
+	/**
+	 * @param date joda DateTime object
+	 * @return Masked DateTime e.g. 1980-01-01
+	 */
+	public DateTime maskDate(DateTime date) {
+		
+		return date != null ? new DateTime(maskDateCore(date.toDate())) : null;
+		
+	}
+	
+	/**
+	 * @param date java LocalDate object
+	 * @return Masked date e.g. 1980-01-01
+	 */
+	public java.time.LocalDate maskDate(java.time.LocalDate date) {
+		
+		if (date != null) {
+			
+			ZoneId defaultZoneId = ZoneId.systemDefault();
+			Date newDate = maskDateCore(Date.from(date.atStartOfDay(defaultZoneId).toInstant()));
+			
+			return Instant.ofEpochMilli(newDate.getTime()).atZone(ZoneId.systemDefault()).toLocalDate();
+			
+		}
+		
+		return null;
+		
+	}
+	
+	/**
+	 * @param T generic date object
+	 * @return Masked Date object
+	 */
+	private <T> Date maskDateCore(T date) {
+		Calendar cal = Calendar.getInstance();
+		cal.setTime((Date) date);
+		
+		cal.set(Calendar.MONTH, 0);
+		cal.set(Calendar.DAY_OF_MONTH, 1);
+		cal.set(Calendar.HOUR_OF_DAY, 0);
+		cal.set(Calendar.MINUTE, 0);
+		cal.set(Calendar.SECOND, 0);
+		cal.set(Calendar.MILLISECOND, 0);
+		
+		Date newDate = cal.getTime();
+		return newDate;
+	}
+	
+	/**
+	 * @param value String
+	 * @return Masked value xxxxxxxxxx
+	 */
+	public String maskString(String value) {
+		
+		return "xxxxxxxxxx";
+		
+	}
+	
+	public List<Client> processDataMasking(List<Client> clients) {
+		
+		for (Client client : clients) {
+			
+			processDataMaskingForClient(client);
+		}
+		
+		return clients;
```

Comment:
Can this return boolean instead since the clients being returned is the same as the input